pcsx2/ps2/pgif.cpp

```diff
@@ -86,983 +33,733 @@ All the PS1 GPU info comes from psx-spx: http://problemkaputt.de/psx-spx.htm
-            pgifReset();
-            ringBufPut(&pgifDatRbC, &data);  //write a reset command for the PS1DRV to get.
-            break;
-        case 0x01:  //Reset Command Buffer (only?)... This would actually mean the Data (GP0()) register buffer.
-            //ringBufClear(&pgifCmdRbC); //as above.
-            ringBufClear(&pgifDatRbC);
-            break;
-        case 0x02:  //Acknowledge GPU IRQ
-            ackGpuIrq();
-            break;
-        case 0x04:  //DMA Direction / Data Request
-            //The PS1DRV doesn't seem to take care of this...
-            //Removing the following two, "damages" the PS logo, so they are probably necessary.
-            //The PS1 program will not start DMA GPU->IOP (only) if this is missing.
-            PGpuStatReg &= ~STAT_DIR_BY_CMD_BITS;               //TODO: Which bits should be controlled here?
-            PGpuStatReg |= (cmd << 29) & STAT_DIR_BY_CMD_BITS;  //0x60000000; //0x20000000 bit 30 is handled by PS1DRV
-            //PGpuStatReg &= ~0x60000000; //TODO: Which bits should be controlled here?
-            //PGpuStatReg |= (cmd <<29) & 0x40000000; //0x60000000; //0x20000000 bit 30 is handled by PS1DRV
-            //0-1  DMA Direction (0=Off, 1=FIFO, 2=CPUtoGP0, 3=GPUREADtoCPU) ;GPUSTAT.29-30
-            //2-23 Not used (zero)
-            drainPgpuDmaLl();  //See comment in this function.
-            break;
-    }
+{
+	//Handles the GP1(10h) command, that requires immediate response.
+	//The data argument is the old data of the register (shouldn't be critical what it contains).
+	switch ((cmd & 0x7))
+	{
+		case 0:
+		case 1:
+		case 6:
+		case 7:
+			break; //Returns Nothing (old value in GPUREAD remains unchanged)
+		case 2:
+			data = pgif.imm_response.reg.e2 & 0x000FFFFF;
+			break; //Read Texture Window setting  ;GP0(E2h) ;20bit/MSBs=Nothing
+		case 3:
+			data = pgif.imm_response.reg.e3 & 0x0007FFFF;
+			break; //Read Draw area top left      ;GP0(E3h) ;19bit/MSBs=Nothing
+		case 4:
+			data = pgif.imm_response.reg.e4 & 0x0007FFFF;
+			break; //Read Draw area bottom right  ;GP0(E4h) ;19bit/MSBs=Nothing
+		case 5:
+			data = pgif.imm_response.reg.e5 & 0x003FFFFF;
+			break; //Read Draw offset             ;GP0(E5h) ;22bit
+	}
+	return data;
+}
+
+void handleGp1Command(u32 cmd)
+{
+	//Check GP1() command and configure PGIF accordingly.
+	//Commands 0x00 - 0x01 are partially handled in ps1drv, we should just clear fifo.
+	u32 cmdNr = ((cmd >> 24) & 0xFF) & 0x3F;
+	u32 param = 0;
+	switch (cmdNr)
+	{
+		case 0:
+			// Pgpu reset, mostly handled by ps1drv (00201D64 in ps1drv). Comment for case 1 apply here too.
+			ringBufferClear(&rb_gp0);
+			break;
+		case 1: //Reset GP0 fifo, seems to check that something is empty, so maybe we should do the same before clear?
+			ringBufferClear(&rb_gp0);
+			break;
+		case 2: //Acknowledge GPU IRQ
+			ackGpuIrq1();
+			break;
+		case 4: //DMA Direction / Data Request. The PS1DRV doesn't care of this... Should we do something on pgif ctrl?
```

Comment:
Could potentially be linked to the SIF2 FIFO direction, however the SIF will also have the direction bits set by the DMA channels also, so this may have no effect at all.  Could be worth noting though.